src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -177,23 +184,36 @@ protected void AssertSelf ()
 			if (!request.RequestUri.IsAbsoluteUri)
 				throw new ArgumentException ("Must represent an absolute URI", "request");
 
-			URL java_url = new URL (request.RequestUri.ToString ());
-			URLConnection java_connection = java_url.OpenConnection ();
-			HttpURLConnection httpConnection = await SetupRequestInternal (request, java_connection);
-			return await ProcessRequest (request, java_url, httpConnection, cancellationToken);
+			var redirectState = new RequestRedirectionState {
+				NewUrl = request.RequestUri,
+				RedirectCounter = 0,
+				Method = request.Method
+			};
+			while (true) {
+				URL java_url = new URL (redirectState.NewUrl.ToString ());
```

Comment:
They are disposed of in the [response message](https://github.com/xamarin/xamarin-android/blob/master/src/Mono.Android/Xamarin.Android.Net/AndroidHttpResponseMessage.cs#L39) and we discard the unneeded response should a redirect happen [right here](https://github.com/grendello/xamarin-android/blob/bug43151/src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs#L254)